client-app/proto.go

```diff
@@ -168,50 +166,24 @@ func (s RandomGen) GetPubkey() ([]byte, error) {
 	return rx[2 : 2+32], nil
 }
 
-func (s RandomGen) GetSignature() ([]byte, error) {
+func (s RandomGen) GetSignature() ([]byte, []byte, error) {
```

Comment:
How about a comment on what it returns? It returns signature, hash, right?

Reply:
Fixed.